lib/vector-core/src/event/value.rs

```diff
@@ -1319,6 +1323,29 @@ mod test {
         Ok(buf)
     }
 
+    mod value_hash {
+        use super::*;
+
+        fn hash(a: Value) -> u64 {
+            let mut h = std::collections::hash_map::DefaultHasher::new();
+
+            a.hash(&mut h);
+            h.finish()
+        }
+
+        #[test]
+        fn hash_correctly() {
+            assert_eq!(hash(Value::Integer(0)), hash(Value::Integer(0)));
+            assert_ne!(hash(Value::Integer(0)), hash(Value::Integer(1)));
+            assert_ne!(hash(Value::Boolean(true)), hash(Value::Integer(2)));
+            assert_eq!(hash(Value::Float(1.2)), hash(Value::Float(1.4)));
+            assert_ne!(hash(Value::Float(-0.0)), hash(Value::Float(0.0)));
+            assert_eq!(hash(Value::Float(f64::NEG_INFINITY)), hash(Value::Float(f64::INFINITY)));
```

Comment:
I neglected infinity 😄 . I think maybe we should have these hash to different values?